src/winrm_client.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import os
+import winrm
+
+
+class WinRMClient:
+    def __init__(self, host, username, password):
+        self.session = winrm.Session(host, auth=(username, password), transport='basic',
+                                     server_cert_validation='ignore')
+
+    def upload_file(self, local_path, remote_path):
```

Comment:
Cognitive complexity of function,  please implement separate functions: to create directory, to check if directory exists, etc.